### DISCUSSION TITLE
"Depends" breaks request processing

### DISCUSSION THREAD

**Discussion by AIshutin** [fastapi/fastapi#7725] [Questions]
### Describe the bug

After adding a dependency to parameter without changing the data format old requests stopped working with error: "422 Unprocessable Entity"

### To Reproduce

1. Create a file "run.py" with:

```Python
import requests
from fastapi import Depends, FastAPI
from pydantic import BaseModel
app = FastAPI()

class Item(BaseModel):
    name: str
    description: str = None
    price: float
    tax: float = None

class User(BaseModel):
    username: str
    full_name: str = None

def dependant(item: Item):
    return {'username': 'username', 'full_name': 'full_name'}

@app.post("/items/{item_id}")
def update_item(item_id: int, user: User, item: Item): #User = Depends(dependant)):
    results = {"item_id": item_id, "item": item, "user": user}
    return results
```
2.  Create a file "run.py" with:
```Python
import requests

sample_body = {
    "item": {
        "name": "Foo",
        "description": "The pretender",
        "price": 42.0,
        "tax": 3.2
    },
    "user": {
        "username": "dave",
        "full_name": "Dave Grohl"
    }
}

print(requests.post('http://localhost:2000/items/1', json=sample_body))
```
3.  Run the server and the test. It works.
4. Change 
```Python
@app.post("/items/{item_id}")
def update_item(item_id: int, user: User, item: Item): #User = Depends(dependant)):
    results = {"item_id": item_id, "item": item, "user": user}
    return results
```
to 
```Python
@app.post("/items/{item_id}")
def update_item(item_id: int, user: User, item: User = Depends(dependant)):
    results = {"item_id": item_id, "item": item, "user": user}
    return results
```
4. Run the server and the test. It does not work.
5. But I expected it to be working like before.

### Expected behavior

I expected, that the change would not affect input format, because the only difference is in internal processing. Also, auto documentation suggests the same.

### Screenshots

![Screenshot from 2020-01-20 21-25-46](https://user-images.githubusercontent.com/30271345/72749705-d79cd180-3bcb-11ea-919c-74cca1386445.png)
![Screenshot from 2020-01-20 21-26-17](https://user-images.githubusercontent.com/30271345/72749703-d7043b00-3bcb-11ea-8221-157280dd9674.png)

### Environment

- OS: Linux
- FastAPI Version 0.46.0
- Python 3.6.9

**Comment by nsidnev** [comment]
@AIshutin 
Hi! Well, I'm not sure that this particular case has been clearly described in the docs, but you're actually looking for [this](https://fastapi.tiangolo.com/tutorial/body-multiple-params/#embed-a-single-body-parameter) page. As described there, `FastAPI` will expect your model to be the root for the JSON body being sent if this model is the only model defined on the route (or dependency in your second case). In the first case, you have already defined 2 models on the same route, so `FastAPI` will expect them as separate fields in JSON.

You can change `FastAPI` behavior with `Body` parameter. In your case, you should be interested in the `embed` argument, which tells `FastAPI` that the model should be expected as a JSON field and not the whole body.

So, if you change your application to this, then everything should work:
```python3
from fastapi import Depends, FastAPI, Body
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str
    description: str = None
    price: float
    tax: float = None


class User(BaseModel):
    username: str
    full_name: str = None


def dependant(item: Item = Body(..., embed=True)):
    return {"username": "username", "full_name": "full_name"}


@app.post("/items/{item_id}")
def update_item(
    item_id: int, user: User = Body(..., embed=True), item: User = Depends(dependant)
):
    results = {"item_id": item_id, "item": item, "user": user}
    return results
```

**Comment by AIshutin** [comment]
Big thanks! You helped a lot.

**Comment by tiangolo** [comment]
Thanks for the help here @nsidnev ! :cake: :bowing_man: 

Thanks @AIshutin for reporting back and closing the issue :+1:

**Comment by YuriiMotov** [comment]
Works fine in current version (fastapi 0.115.14, pydantic 2.11.7)

```python
@app.post("/items/{item_id}")
def update_item(item_id: int, user: User, item: User = Depends(dependant)):
    results = {"item_id": item_id, "item": item, "user": user}
    return results
```

![image](https://github.com/user-attachments/assets/ae0fbd2b-3a33-4ef1-90ef-a1835446a837)
